Library/Homebrew/bottle_checksum.rb

```diff
@@ -0,0 +1,33 @@
+# typed: true
+# frozen_string_literal: true
+
+# A formula's checksum.
+#
+# @api private
+class BottleChecksum
+  extend Forwardable
+
+  attr_reader :hash_type, :hexdigest, :tag, :cellar
+
+  TYPE = :sha256
+
+  def initialize(hexdigest, tag = nil, cellar = nil)
+    @hash_type = TYPE
+    @hexdigest = hexdigest.downcase
+    @tag = tag
+    @cellar = cellar
+  end
+
+  delegate [:empty?, :to_s, :length, :[]] => :@hexdigest
+
+  def ==(other)
+    case other
```

Comment:
Unit tests would be nice for this method.

Reply:
I made BottleChecksum a child class of Checksum, this method is not needed anymore. Checksum comparison can directly be done by the parent method.